### PR TITLE
docs: clarify docs around endpoints and metadata server

### PR DESCRIPTION
**Pull request by smira**
Bump references to Talos, remove explicit `Metadata Server` component as
it got merged into `metal-controller-manager`.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: docs/website/content/docs/v0.3/Configuration/environments.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ metadata:
   name: default
 spec:
   kernel:
-    url: "https://github.com/talos-systems/talos/releases/download/v0.8.1/vmlinuz-amd64"
+    url: "https://github.com/talos-systems/talos/releases/download/v0.10.2/vmlinuz-amd64"
```
Comment (AlekSi) [inline docs/website/content/docs/v0.3/Configuration/environments.md:30]: I bumped those to 0.9.3 in my PR because that's what we use for testing

**Review by andrewrynhard — APPROVED** [review]

File: docs/website/content/docs/v0.3/Configuration/environments.md (hunk 1 above)
Comment (smira) [reply docs/website/content/docs/v0.3/Configuration/environments.md:30]: let me bump our version to 0.10.2 then, I'll do that in the separate PR

we need Sidero to keep up to date with Talos!

**Pull request merged**